File: app/src/main/java/org/alkaline/taskbrain/ui/currentnote/selection/EditorSelection.kt
package org.alkaline.taskbrain.ui.currentnote.selection

/**
 * Represents a selection range in the editor using global character offsets.
 */
data class EditorSelection(
    val start: Int,
    val end: Int
) {
    val min: Int get() = minOf(start, end)
    val max: Int get() = maxOf(start, end)
    val isCollapsed: Boolean get() = start == end
    val hasSelection: Boolean get() = start != end

    companion object {
        val None = EditorSelection(-1, -1)
    }
}

File: web/src/editor/EditorSelection.ts
export interface EditorSelection {
  start: number
  end: number
}

export const SELECTION_NONE: EditorSelection = { start: -1, end: -1 }

export function selMin(sel: EditorSelection): number {
  return Math.min(sel.start, sel.end)
}

export function selMax(sel: EditorSelection): number {
  return Math.max(sel.start, sel.end)
}

export function hasSelection(sel: EditorSelection): boolean {
  return sel.start !== sel.end && sel.start >= 0
}
